src/arrayMethodReduce.js

```diff
@@ -5,7 +5,17 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator = initialValue;
+
+    if (arguments.length < 2) {
+      accumulator = this[0];
+    }
+
+    for (let i = accumulator !== this[0] ? 0 : 1; i < this.length; i++) {
+      accumulator = callback(accumulator, this[i], i, this);
+    }
```

Comment:
let's improve readability here

Reply:
as I see now - you can create a variable for the condition you're trying to get here